src/main/java/org/shanerx/tradeshop/TradeShop.java

```diff
@@ -49,364 +49,377 @@
+		toSave += addMessage("no-ts-open", "&cThat TradeShop does not belong to you");
+		toSave += addMessage("empty-ts-on-setup", "&cTradeShop empty, please remember to fill it!");
+		toSave += addMessage("on-trade", "&aYou have traded your&e {AMOUNT2} {ITEM2} &a for &e {AMOUNT1} {ITEM1} &awith {SELLER}");
+		toSave += addMessage("insufficient-items", "&cYou do not have &e {AMOUNT} {ITEM}&c!");
+		toSave += addMessage("shop-full-amount", "&cThe shop does not have &e{AMOUNT} &cof a single type of &e{ITEM}&c!");
+		toSave += addMessage("full-amount", "&cYou must have &e{AMOUNT} &cof a single type of &e{ITEM}&c!");
+		toSave += addMessage("shop-empty", "&cThis TradeShop does not have &e {AMOUNT} {ITEM}&c!");
+		toSave += addMessage("shop-full", "&cThis TradeShop is full, please contact the owner to get it emptied!");
+		toSave += addMessage("player-full", "&cYour inventory is full, please make room before trading items!");
+		toSave += addMessage("confirm-trade", "&eTrade &6 {AMOUNT1} {ITEM1} &e for &6 {AMOUNT2} {ITEM2} &e?");
+		toSave += addMessage("held-item", "\n&6You are currently holding: " +
+				"\n&2Material: &e{MATERIAL}" +
+				"\n&2ID Number: &e{ID}" +
+				"\n&2Durability: &e{DURABILITY}" +
+				"\n&2Amount: &e{AMOUNT}" +
+				"\n&6You do not need to use the durability if it is 0" +
+				"\n&6Put this on your TradeShop sign: " +
+				"\n&e{AMOUNT} {MATERIAL}:{DURABILITY} " +
+				"\n&e{AMOUNT} {ID}:{DURABILITY}");
+		toSave += addMessage("held-empty", "&eYou are currently holding nothing.");
+		toSave += addMessage("player-only-command", "&eThis command is only available to players.");
+		toSave += addMessage("missing-shop", "&cThere is not currently a shop here, please tell the owner or come back later!");
+		toSave += addMessage("no-sighted-shop", "&cNo shop in range!");
+		toSave += addMessage("updated-shop-members", "&aShop owners and members have been updated!");
+		toSave += addMessage("unsuccessful-shop-members", "&aThat player is either already on the shop, or you have reached the maximum number of users!");
+		toSave += addMessage("who-message", "&6Shop users are:\n&2Owners: &e{OWNERS}\n&2Members: &e{MEMBERS}");
+		toSave += addMessage("self-owned", "&cYou cannot buy from a shop in which you are a user.");
+		toSave += addMessage("not-owner", "&cYou cannot create a sign for a shop that you do not own.");
+		toSave += addMessage("illegal-item", "&cYou cannot use one or more of those items in shops.");
+		toSave += addMessage("missing-item", "&cYour sign is missing an item for trade.");
+		toSave += addMessage("missing-info", "&cYour sign is missing necessary information.");
+		toSave += addMessage("amount-not-num", "&cYou should have an amount before each item.");
+		toSave += addMessage("buy-failed-sign", "&cThis shop sign does not seem to be formatted correctly, please notify the owner.");
+
+		if (toSave > 0)
+			save();
+	}
+
+	private void addSettingsDefaults() {
```

Comment:
Change Here

Reply:
takes all the returns from the adds and saves only if something was changed(value greater then 0)